docs/scripts/formattedTSDemos.js

```diff
@@ -1,55 +1,99 @@
+/* eslint-disable no-console */
+/* eslint-disable no-continue */
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-restricted-syntax */
+
 /**
  * Transpiles and formats TS demos.
  * Can be used to verify that JS and TS demos are equivalent. No introduced change
  * would indicate equivalence.
  */
-const childProcess = require('child_process');
+
+/**
+ * List of demos to ignore when transpiling
+ * Example: "app-bar/BottomAppBar.tsx"
+ */
+const ignoreList = [];
+
 const fse = require('fs-extra');
 const path = require('path');
+const babel = require('@babel/core');
 const prettier = require('prettier');
-const util = require('util');
+const os = require('os');
+
+const babelConfig = {
+  presets: ['@babel/preset-typescript'],
+  plugins: ['unwrap-createStyles'],
+  generatorOpts: { retainLines: true },
+  babelrc: false,
+  configFile: false,
+};
+
+const prettierConfig = prettier.resolveConfig(process.cwd(), {
+  config: path.join(__dirname, '../../prettier.config.js'),
+});
+
+const watchMode = process.argv.some(arg => arg === '--watch');
 
-const exec = util.promisify(childProcess.exec);
+async function getFiles(root) {
+  const files = [];
+  const folderItem = await fse.readdir(root);
 
-async function getUnstagedGitFiles() {
-  const { stdout } = await exec('git diff --name-only');
-  const list = stdout.trim();
+  for (const name of folderItem) {
+    const filePath = path.join(root, name);
+    const stat = await fse.stat(filePath);
 
-  if (list === '') {
-    // "".split(" ") => [""]
-    return [];
+    if (stat.isFile()) {
+      if (!filePath.endsWith('.tsx')) continue;
+      if (ignoreList.some(ignorePath => filePath.endsWith(path.normalize(ignorePath)))) continue;
+      files.push(filePath);
+    } else if (stat.isDirectory()) {
+      files.push(...(await getFiles(filePath)));
+    }
   }
 
-  return list.split('\n');
+  return files;
 }
 
+const fixBabelIssuesRegExp = new RegExp(/(?<=(\/>)|,)(\r\n|\r|\n){2}/g);
 function fixBabelGeneratorIssues(source) {
-  return source.replace(/\/>\n\n/g, '/>\n').replace(/,\n\n/g, ',\n');
+  return source.replace(fixBabelIssuesRegExp, os.EOL);
```

Comment:
I think we should just scan the input for the first linefeed and use that one. I think this can cause issues if you checkout `\n` on windows.

Reply:
https://git-scm.com/docs/gitattributes#_checking_out_and_checking_in
Wont git always checkout with `\r\n` on Windows because of `* text=auto`?

Reply:
What happens if I have configured this globally?

Reply:
Quoting the documentation at https://git-scm.com/docs/gitattributes
>When deciding what attributes are assigned to a path, Git consults $GIT_DIR/info/attributes file (which has the highest precedence), .gitattributes file in the same directory as the path in question, and its parent directories up to the toplevel of the work tree (the further the directory that contains .gitattributes is from the path in question, the lower its precedence). Finally global and system-wide files are considered (they have the lowest precedence).

So it is possible for someone to change it, I'll make it use the first linefeed it finds

Reply:
It now uses the first line ending it finds, if none is found, it defaults to `os.EOL`